### PR TITLE
HTTPS/HSTS by default

### PR DESCRIPTION
**Pull request by EdwardAngert**
duplicate of #4047 
made to merge into Master before Gatsby, if needed

### CODE REVIEW

File: source/_docs/pantheon-yml.md [hunk 1]
```diff
@@ -36,6 +36,32 @@ protected_web_paths:
 * You may not be able to protect files or paths with special characters
 * Wait a few seconds for changes to take effect
 
+### Enforce HTTPS + HSTS
```
Comment (joshkoenig) [inline source/_docs/pantheon-yml.md]: The markup in this section is broken. It's not rendering as a list.

Comment (EdwardAngert) [reply source/_docs/pantheon-yml.md]: thanks, @joshkoenig! this is fixed in my most recent push
check it out on the pre-built here https://httpsnong-static-docs.pantheonsite.io/docs/pantheon-yml/#enforce-https-+-hsts

File: source/_docs/pantheon-yml.md [hunk 2]
```diff
@@ -36,6 +36,36 @@ protected_web_paths:
...
+
+HTTPS adds a layer of encryption that prevents others from snooping on or tampering with traffic to your site. HTTP Strict Transport Security (**HSTS**) instructs browsers to only connect via HTTPS and helps protect websites against protocol downgrade attacks and cookie hijacking.
+
+<div class="alert alert-info" role="alert">
+<h4 class="info">Note</h4>
+<p markdown="1">Before adjusting `enforce_https`, review and understand the configuration options and all considerations to avoid unintended consequences.</p>
+</div>
+
+Ensure that your site will always use HTTPS to deliver content with `enforce_https`. Five values are available, from least to most secure:
```
Comment (rachelwhitton) [inline source/_docs/pantheon-yml.md]: Suggest adding the following block: 

```
enforce_https: full+subdomains
``` 

^ demonstrate valid syntax as a key-value pair (vs an array) and align new section w/existing (all other configs have a code example nearby)

File: source/_docs/pantheon-yml.md [hunk 3]
```diff
@@ -36,6 +36,36 @@ protected_web_paths:
...
+
+- `off`
```
Comment (rachelwhitton) [inline source/_docs/pantheon-yml.md]: Suggest adding expected behavior for `off`, and note this is the default config

File: source/_docs/pantheon-yml.md [hunk 4]
```diff
@@ -36,6 +36,36 @@ protected_web_paths:
...
+- `full` - Redirect to HTTPS with a year-long HSTS header
+- `full+subdomains` - Redirect to HTTPS with a year-long HSTS header that will enforce HTTPS for subdomains, even those not on Pantheon.
+   - This is the recommended and most secure configuration.
+
+#### Considerations
+* Use of `full` or `full+subdomains` should be treated as a commitment. If your site is unable to serve HTTPS (e.g. by moving to a host that doesn't support HTTPS) after sending a long-duration HSTS header, visitors will be unable to access your site.
+* Any option with `+subdomains` should only be used if you want to enforce HTTPS for *all subdomains, even those not connected to Pantheon*.
+* To prepare your site to serve all content via HTTPS, follow the [Switching Sites from HTTP to HTTPS](/docs/http-to-https/) doc.
+
+#### Test Your Site's HSTS Configuration for an A+ Rating
```
Comment (rachelwhitton) [inline source/_docs/pantheon-yml.md]: Most other sections are much shorter, and link out to a fuller doc page with additional context (e.g., Drush, PHP, Nested Docroot, etc.) 

Is this section better served by `/docs/http-to-https` ?

File: source/_partials/content/hsts.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+<li>HTTPS redirect with a long HSTS duration (366 days, for an A+ SSL Labs rating), which does not apply to subdomains:
+
+    <pre><code class="hljs less">scheme: full</code></pre></li>
+
+<li>HTTPS redirect with a long HSTS duration (366 days, for an A+ SSL Labs rating), which also applies to subdomains:
+
+    <pre><code class="hljs less">scheme: full+subdomains</code></pre></li>
+</ul>
+
+Pantheon will send the following header based on the setting of <code>enforce_https</code>:
```
Comment (rachelwhitton) [inline source/_partials/content/hsts.html]: Suggest something like: 

Pantheon configures the `Strict-Transport-Security` header according to the configuration of `enforced_https`: 

- `off` (default): HSTS is disabled, `Strict-Transport-Security` header is not set by Pantheon, HTTP will not forcefully redirect to HTTPS
- `transitional`: HSTS is enabled, `Strict-Transport-Security: max-age=300` is set by Pantheon, HTTP will redirect forcefully redirect to HTTPS
...etc

Comment (EdwardAngert) [reply source/_partials/content/hsts.html]: thanks @rachelwhitton !
I tried to work your suggestions in wherever I could in the new new PR #4998

**Comment by EdwardAngert** [comment]
closed in favor of #4998

**Pull request closed without merging**